sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Monitor.OpenTelemetry.Exporter;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace Azure.Monitor.OpenTelemetry
+{
+    internal class AzureMonitorOpenTelemetryImplementations
+    {
+        internal static IServiceCollection AddAzureMonitorOpenTelemetryWithOptions(IServiceCollection services, AzureMonitorOpenTelemetryOptions? options)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            options ??= new AzureMonitorOpenTelemetryOptions();
+            var builder = services.AddOpenTelemetry();
+
+            if (options.EnableTraces)
+            {
+                builder.WithTracing(b => b
+                             .AddAspNetCoreInstrumentation()
+                             .AddAzureMonitorTraceExporter(o =>
+                             {
+                                 o.ConnectionString = options.ConnectionString;
+                                 o.DisableOfflineStorage = options.DisableOfflineStorage;
+                                 o.StorageDirectory = options.StorageDirectory;
+                             }));
+            }
+
+            if (options.EnableMetrics)
+            {
+                builder.WithMetrics(b => b
+                             .AddAspNetCoreInstrumentation()
+                             .AddAzureMonitorMetricExporter(o =>
+                             {
+                                 o.ConnectionString = options.ConnectionString;
```

Comment:
I didn't read it line by line, is options already an instance of AzureMonitorOpenTelemetryOptions which in turn is a child class of AzureMonitorExporterOptions? If that's the case could we directly pass it? (e.g. `AddAzureMonitorMetricExporter(options)`)

Reply:
`AzureMonitorOpenTelemetryOptions` is inherited from `AzureMonitorExporterOptions`.

`public class AzureMonitorOpenTelemetryOptions : AzureMonitorExporterOptions`

When we plan to switch completely to OTLP, we could copy all the properties from ExporterOptions to AzureMonitorOpenTelemetryOptions.